### ISSUE TITLE
Cannot use the FAIL log level if test itself does not FAIL

### ISSUE THREAD

**Issue by joaonc** [robotframework/statuschecker#4]
I'm trying to update the test case [Page Should Contain](https://github.com/robotframework/SeleniumLibrary/blob/bc6a4af75081cd2f59a690af495d213f2d7cbc39/atest/acceptance/keywords/content_assertions.robot#L51) with
```
Page Should Contain
    [Documentation]    Default log level does not have html output.
    ...    LOG 2:7 Current page contains text 'needle'.
    ...    LOG 4.1:14 FAIL REGEXP: .*
    Page Should Contain    needle
    Page Should Contain    This is the haystack
    Run Keyword And Expect Error
    ...    Page should have contained text 'non existing text' but did not.
    ...    Page Should Contain    non existing text
```

Basically, my [code](https://github.com/robotframework/SeleniumLibrary/pull/1244/commits/6c75d015bb4465afd168ac138988dd17692c153c#diff-530ec8e37846ff25bdd55df7c643b806R108) is to make it so that keyword does _not_ log the DOM at INFO and that test case runs at INFO level and should check that the DOM is indeed not logged.

To check that the DOM is not logged, it needs to check line 14 of step 4.1.
So I need to check that that line has status `FAIL`.

`LOG 4.1:14 FAIL REGEXP: .*` indeed checks that that line has status `FAIL` (`LogMessageChecker._check_msg_level(...)` passes), but also uses that information to check that the **test case** has status `FAIL` (`TestStatusChecker._check_status(...)` fails).

Looks like `TestStatusChecker._check_status(...)` should **not** use the values from `LOG` to check for test case status.

![image](https://user-images.githubusercontent.com/11450769/49118580-b6f01a80-f259-11e8-95f0-2f53f8f5795c.png)

**Comment by joaonc** [comment]
Note: Another way to check that the DOM is _not_ there, would be to add
`LOG 4.1:14 REGEXP: (?i)^(?!<html).*$`  (this regex matches when the line doesn't start with `'<html'`, case insensitive)

But that fails at `LogMessageChecker._check_msg_level(...)`.

Would be good to have a `NONE` option to ignore log level, something like:
`LOG 4.1:14 NONE REGEXP: (?i)^(?!<html).*$`

**Comment by pekkaklarck** [comment]
Could you try adding `PASS` into the beginning of the doc to explicitly specify that the test should pass? The reminder of the doc ought to then be used for validating log messages.

**Comment by joaonc** [comment]
That didn't work b/c [here](https://github.com/robotframework/statuschecker/blob/7b59e30bbddf7740a010a14da61c3a4e99c6bc2f/robotstatuschecker.py#L99) we have

```
def _get_status(self, doc):	    def _get_status(self, doc):
        return 'FAIL' if 'FAIL' in doc else 'PASS'
```

Which returns `FAIL` if we have `LOG 4.1:14 FAIL The failure message` but the test case passes.

Created a fix for this in PR #5 but there's something I can't point my finger in the test cases that verify this functionality..

**Comment by pekkaklarck** [comment]
You are right, if there's `FAIL` anywhere the expected status is `FAIL` even if there would be `PASS` as well. Y

**Comment by pekkaklarck** [comment]
If I've understood this issue correctly, it only manifests itself if the test itself is expected to pass but there are log messages with `FAIL` level. This can happen, for example, when using 'Run Keyword And Expect Error` to validate that the expected error occurs. That's definitely a bug, but in many cases it can be avoided by not using `Run Keyword And Expect Error` (i.e. letting the test fail) and adding the needed `FAIL Whatever the message is` into the documentation.

**Comment by pekkaklarck** [comment]
There's no log level `PASS` so the problem only occurs if test is expected to pass but to have `FAIL` messages. This obviously needs to be fixed, but because there's a workaround I don't consider the fix (and the needed release) urgent.

**Comment by joaonc** [comment]
> There's no log level `PASS`

Right, my bad on the previous comment.
These test cases were name appropriately:
- `Test case PASS log INFO`
- `Test case FAIL log INFO`

I'll try to get the PR in _SeleniumLibrary_ working with the workaround.

**Comment by pekkaklarck** [comment]
This is an annoying limitation and it makes sense to fix it. The approach in PR #5 looks fine, but due to recent changes to tests I probably cannot merge it directly. Probably I'll just copy relevant parts from the PR or use it as an example.

Thanks anyway for the PR and the report @joaonc. This isn't too actively maintained project, but better late than never I guess...
